src/seqn/seqn.lisp

```diff
@@ -0,0 +1,139 @@
+(in-package :geb.seqn.main)
+
+(defun fill-in (num seq)
+  "Fills in extra inputs on the right with 0-oes"
+  (geb.utils:apply-n num (lambda (x) (append x (list 0))) seq))
+
+(defun 0-list (n)
+  (if (= n 1)
+      (list 0)
+      (cons 0 (0-list (- n 1)))))
+
+(defun prod-list (l1 l2)
+  "Takes two lists of same length and gives pointwise product
+where first element come from first list and second from second
+
+```lisp
+SEQN> (prod-list (list 1 2) (list 3 4))
+((1 3) (2 4))
+```"
+  (if (or (null l1)
+          (null l2))
+      nil
+      (let ((lh1 (length l1))
+            (lh2 (length l2)))
+        (if (= lh1 1)
+            (list (list (car l1) (car l2)))
+            (append (prod-list (butlast l1 1)
+                               (butlast l2 1))
+                    (list
+                     (list (nth (1- lh1) l1) (nth (1- lh2) l2))))))))
+
+(defun seq-max-fill (seq1 seq2)
+  "Takes two lists, makes them same length by adding 0es on the right
+where necessary and takes their pointwise product"
+  (let ((l1 (length seq1))
+        (l2 (length seq2)))
+    (if (< l1 l2)
+        (prod-list (fill-in (- l2 l1) seq1) seq2)
+        (prod-list seq1 (fill-in (- l1 l2) seq2)))))
+
+(defun max-list (lst)
+  (max (car lst) (car (cdr lst))))
+
+(defmethod seqwidth ((obj <substobj>))
+  (typecase-of substobj obj
+    (so0 (list 0))
+    (so1 (list 0))
+    (coprod (cons 1 (mapcar #'max-list
+                            (seq-max-fill
+                             (seqwidth
+                              (mcar obj))
+                             (seqwidth
+                              (mcadr obj))))))
```

Comment:
reading this makes it a bit hard to read, formatting it like this is more obvious to my eyes at least

```lisp
    (coprod (cons 1
                  (mapcar #'max-list
                          (seq-max-fill (seqwidth (mcar obj))
                                        (seqwidth (mcadr obj))))))
```